### ISSUE TITLE
Unable to find app in conjunction with Wagtail 404 URL

### ISSUE THREAD

**Issue by hubbs5** [GibbsConsulting/django-plotly-dash#386]
I am trying to add a Dash app to a Wagtail/CoderedCMS site, but I cannot get the URL for the Dash app to be found. I followed the instructions in the documentation as well as other threads here, but I'm stuck with this. It's probably a simple issue, but I'm just not finding the solution.

Initialize site with [CoderedCMS](https://github.com/coderedcorp/coderedcms) (`pip install coderedcms` first):
```
coderedcms start mysite --sitename mysite --domain www.example.com
cd mysite/
python manage.py migrate
python manage.py createsuperuser
python manage.py runserver
```

Site is running. Now add the simple app from the Django Plotly Dash example. From `mysite/`
```
mkdir apps
cd apps
mkdir django_plotly_dash
cd django_plotly_dash
touch app.py
```
[Copy example app](https://django-plotly-dash.readthedocs.io/en/latest/simple_use.html) into `app.py`.

Create `urls.py`:

```
# urls.py
from django.urls import path
from . import views

app_name = "SimpleExample"
urlpatterns = [
    path("example", views.example, name="example")
]
```

Create `views.py`:

```
# views.py
from django.shortcuts import render

def example(request, **kwargs):
    context = {}
    template_name = "simple.html"

    return render(request, template_name=template_name, context=context)
```
Navigate to `mysite/website/templates/` and create `simple.html`:

```
{% load plotly_dash %}

{% plotly_app name="SimpleExample"}
```

Now modify the settings and urls files.

Go to `mysite/mysite/urls.py` and add `path('django_plotly_dash/', include('apps.django_plotly_dash.urls')),` to the `urlpatterns` list.

Go to `mysite/mysite/settings/base.py` add `django_plotly_dash.apps.DjangoPlotlyDashConfig` to `INSTALLED_APPS`.
Add `X_FRAME_OPTIONS = 'SAMEORIGIN'`.

From `mysite/` run:
```
python manage.py makemigrations
python manage.py migrate
```

Go to http://127.0.0.1:8000/django_plotly_dash/example/ (restart server if needed) and get 404 error.

New files and structure:
mysite
--apps
----django_plotly_dash
------urls.py
------app.py
------views.py

Most frustrating is that I ran this once, and it worked, at least I saw the template but no app. I removed the `django_plotly_dash` from the path to see if I could get it from just http://127.0.0.1:8000/example/ and I couldn't get it to revert, let alone display the app. I feel like I tried every variation of imports, urls, etc. and couldn't get around this, so any help would be greatly appreciated!

Running locally with:
Python 3.9.7
MacOS Monterey 12.2.1
django==3.2.12
dash==2.2.0
django-plotly-dash==1.6.6
coderedcms==0.22.3

**Comment by GibbsConsulting** [comment]
@hubbs5 it looks like you're not importing the app (in `app.py` based on your comprehensive description above) anywhere. Importing it in your urls.py should suffice.

Are you able to determine what is causing the server to generate the 404 response? I would expect that it is due to the dash app not being located as it has not been imported.

**Comment by hubbs5** [comment]
Thanks for the response.

I added `from apps.django_plotly_dash.app import app` to the `mysite/mysite/urls.py` file and added `from .app import app` to the `mysite/apps/django_plotly_dash/urls.py` file and continue to get the same error (with multiple `urls.py` files, I just tried them all).

The 404 error simply states: `Not Found: /django_plotly_dash/example/`

<img width="1015" alt="image" src="https://user-images.githubusercontent.com/5096273/157247910-3cc284e7-13a1-43cb-a055-5d11d22a2b40.png">

**Comment by GibbsConsulting** [comment]
You need to add `django_plotly_dash` to your `INSTALLED_APPS` into the settings file that is being used, and then ensure that the root project `urls.py` file is including the url patterns. Whilst you state above that you have added the app, according to that error the urls are not present in the list of urls being checked by Django. The next thing to check would be the content of `mysite.urls`

**Comment by GibbsConsulting** [comment]
@hubbs5 that last url (wagtail_serve) looks quite greedy - you should also check that any url paths you've added are ahead of that in the root url config

**Comment by hubbs5** [comment]
I was able to recreate the [example here](https://www.youtube.com/watch?v=psvU4zwO3Ao) (which helped me better understand Django) and saw that I was indeed missing importing the app into `urls.py` as you stated. I have `'django_plotly_dash.apps.DjangoPlotlyDashConfig'` in my `INSTALLED_APPS` and moved the order as suggested, but no luck. Same error.

**Comment by hubbs5** [comment]
Do you know how I should change it if they haven't been added successfully? I'm completely stumped by this. I thought it would be rather straightforward (I can add other Django apps to our Wagtail CMS, just not this).

**Comment by hubbs5** [comment]
Yes, that's correct.

**Comment by GibbsConsulting** [comment]
OK, so does the list of url matches tried (from the error message, similar to above) have entries that match those of the urls.py file? Perhaps sharing both the `urlpatterns` part of the urls.py file, along with the 'Django tried these URL patterns' part of the error message, would help here.

**Comment by hubbs5** [comment]
I got it to work!

Instead of starting with a Wagtail app, I started with the Django plotly app from the [tutorial I shared before]([example here](https://www.youtube.com/watch?v=psvU4zwO3Ao) and was able to add a Wagtail app alongside it (following the docs here](https://docs.wagtail.org/en/v2.16.1/getting_started/integrating_into_django.html).

Still not sure why the approach I was taking before failed so miserably, but at least I have a path forward. Thanks for the support.
